src/main/java/com/linkedin/kmf/apps/MirrorPipelineMonitor.java

```diff
@@ -0,0 +1,410 @@
+ */
+package com.linkedin.kmf.apps;
+
+import com.linkedin.kmf.apps.configs.MirrorPipelineMonitorConfig;
+import com.linkedin.kmf.services.ConsumeService;
+import com.linkedin.kmf.services.DefaultMetricsReporterService;
+import com.linkedin.kmf.services.JettyService;
+import com.linkedin.kmf.services.JolokiaService;
+import com.linkedin.kmf.services.ProduceService;
+import com.linkedin.kmf.services.TopicManagementService;
+import com.linkedin.kmf.services.configs.CommonServiceConfig;
+import com.linkedin.kmf.services.configs.ConsumeServiceConfig;
+import com.linkedin.kmf.services.configs.DefaultMetricsReporterServiceConfig;
+import com.linkedin.kmf.services.configs.ProduceServiceConfig;
+import com.linkedin.kmf.services.configs.TopicManagementServiceConfig;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.sourceforge.argparse4j.ArgumentParsers;
+import net.sourceforge.argparse4j.inf.ArgumentParser;
+import net.sourceforge.argparse4j.inf.Namespace;
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static net.sourceforge.argparse4j.impl.Arguments.store;
+
+/*
+ * The MirrorPipelineMonitor app is intended to monitor the health and performance of kafka brokers across a mirror maker
+ * pipeline. It creates one producer and one consumer with the given configuration, produces messages with increasing
+ * integer in the message String, consumes messages, and keeps track of number of lost messages, duplicate messages, e2e
+ * latency, throughput, etc. As well, MirrorPipelineMonitor creates a topic management service for every kafka and zookeeper
+ * broker in the pipeline.
+ *
+ * MirrorPipelineMonitor app exports these metrics via JMX. It also periodically report metrics if DEBUG level logging
+ * is enabled. This information can be used by other application to trigger alert when kafka brokers fail. It also
+ * allows users to track pipeline performance, e.g. latency and throughput.
+ */
+
+public class MirrorPipelineMonitor implements App {
+  private static final Logger LOG = LoggerFactory.getLogger(MirrorPipelineMonitor.class);
+
+  private final ProduceService _produceService;
+  private final ConsumeService _consumeService;
+  private final List<TopicManagementService> _topicManagementServices;
+  private final List<String> _zookeeperUrls;
+  private final String _name;
+
+  public MirrorPipelineMonitor(Map<String, Object> props, String name) throws Exception {
+    _name = name;
+    MirrorPipelineMonitorConfig config = new MirrorPipelineMonitorConfig(props);
+    _zookeeperUrls = config.getList(MirrorPipelineMonitorConfig.ZOOKEEPER_CONNECT_LIST_CONFIG);
+    _produceService = new ProduceService(createProduceServiceConfig(props), name);
+    _consumeService = new ConsumeService(createConsumeServiceConfig(props), name);
+    _topicManagementServices = new ArrayList<>();
```

Comment:
How about `_topicManagementServices = createTopicManagementServices(props, name)`?

Also, maybe rename `createProduceServiceConfig` to `createProduceServiceProps` and `createConsumeServiceConfig` to `createConsumeServiceProps`. Same for `createTopicManagementServiceConfig`.